hw3/02-charts.js

```diff
@@ -0,0 +1,89 @@
+const backgroundColors = [
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
+];
+
+const borderColors = [
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
+];
+
+const url = "https://thronesapi.com/api/v2/Characters";
+
+async function getCharacters() {
+  const response = await fetch(url);
+  const characters = await response.json();
+  return characters;
+}
+
+async function renderChart() {
+  const characters = await getCharacters();
+  const houseRegex = /House /;
+  const houseCounts = {};
+
+  characters.forEach((character) => {
+    let house = character.family;
+
+    if (
+      house === "House Targaryn" ||
+      house === "Targaryn" ||
+      house === "Targaryan"
+    ) {
+      house = "House Targaryen";
+    } else if (
+      house === "House Lannister" ||
+      house === "Lanister" ||
+      house === "Lannister"
+    ) {
+      house = "House Lanister";
+    } else if (
+      house === "" ||
+      house === "Unknown" ||
+      house === "Unkown" ||
+      house === "None"
+    ) {
+      houseCounts["Unknown"] = (houseCounts["Unknown"] || 0) + 1;
+    } else {
+      const familyName = house.replace(houseRegex, "");
+      houseCounts[familyName] = (houseCounts[familyName] || 0) + 1;
+    }
+  });
+
+  const donutChart = document.querySelector(".donut-chart");
+
+  new Chart(donutChart, {
+    type: "doughnut",
+    data: {
+      labels: Object.keys(houseCounts),
+      datasets: [
+        {
+          label: "Number of Characters",
+          data: Object.values(houseCounts),
+          backgroundColor: backgroundColors,
+          borderColor: borderColors,
```

Comment:
![sai1chart](https://github.com/KSKRLIFE/Frontend-HW/assets/38127011/8dada917-b4af-4a24-81fe-4ed6b526d0c5)

The styling of the chart can certainly be improved. Consider adding a background color for each slice of the chart (house).